nixos/modules/services/networking/headscale.nix

```diff
@@ -489,6 +493,8 @@ in {
         SystemCallFilter = ["@system-service" "~@privileged" "@chown"];
         SystemCallArchitectures = "native";
         RestrictAddressFamilies = "AF_INET AF_INET6 AF_UNIX";
+      } // lib.optionalAttrs (cfg.settings.oidc.client_secret_file != null) {
+        LoadCredential = "hs_client_oidc_secret:${cfg.settings.oidc.client_secret_file}";
```

Comment:
I am not completely on top of this systemd cred store, so I will ask silly questions:

Here you tell systemd to fetch the configuration from disk, where for example people put it with `age` or one of the other secrets managers?

Reply:
systemd can only see files or, as of writing, encrypted files. The encryption key may be in a TPM2 device or on disk. See more in https://systemd.io/CREDENTIALS/#encryption

AFAIK systemd does not provide support for other secrets managers.